Main.elm

```diff
@@ -108,23 +127,77 @@ update message model =
             { model | lock = True, content = "", passphrase = "", error = "Wrong passphrase" } ! []
 
         Lock ->
-            { model | lock = True, content = "", passphrase = "" } ! [ setData model.content ]
+            { model | lock = True, content = "", passphrase = "" } ! [ encryptData { content = model.content, passphrase = model.passphrase } ]
 
-        DataSaved _ ->
-            { model | modified = False } ! []
-
-        DataNotSaved error ->
-            { model | error = (Debug.log "" error) } ! []
+        DataSaved result ->
+            let
+                _ =
+                    Debug.log "kinto result" result
+            in
+                { model | modified = False } ! []
 
         ToggleReveal ->
             { model | reveal = not model.reveal } ! []
 
         TimeOut debounceCount ->
             if debounceCount == model.debounceCount then
-                { model | debounceCount = 0 } ! [ setData model.content ]
+                { model | debounceCount = 0 } ! [ encryptData { content = model.content, passphrase = model.passphrase } ]
             else
                 model ! []
 
+        DataEncrypted encrypted ->
+            { model | encryptedData = Debug.log "encrypted data from js" <| Just encrypted } ! [ saveData model.email model.passphrase encrypted ]
+
+        DataNotEncrypted error ->
+            { model | error = (Debug.log "" error) } ! []
+
+
+
+-- Kinto related
+
+
+recordResource : Kinto.Resource String
+recordResource =
+    Kinto.recordResource
+        "default"
+        "hoverpad"
```

Comment:
We should make sure we are not leeking privacy information in the database. This ID should be hashed with the user password.